include/PICA/pica_frag_config.hpp

```diff
@@ -29,6 +29,18 @@ namespace PICA {
 		std::array<u32, 4 * 6> tevConfigs;
 	};
 
+	struct FogConfig {
+		union {
+			u32 raw{};
+
+			BitField<0, 3, FogMode> mode;
+			BitField<3, 1, u32> flipDepth;
+			BitField<8, 8, u32> fogColorR;
+			BitField<16, 8, u32> fogColorG;
```

Comment:
For texenv constant colours it's normally ideal to pass them as uniforms as they change constantly, though we'll see if this is necessary here, since I don't expect games to constantly change fog colour